custom/icds_reports/utils/aggregation_helpers/distributed/agg_awc.py

```diff
@@ -69,31 +69,33 @@ def aggregation_query(self):
             1,
             'no',
             5,
-            CASE WHEN
-                (count(*) filter (WHERE date_trunc('MONTH', vhsnd_date_past_month) = %(start_date)s))>0
-                THEN 1 ELSE 0 END,
-            CASE WHEN
-                (count(*) filter (WHERE date_trunc('MONTH', date_cbe_organise) = %(start_date)s))> 1
-                THEN 1 ELSE 0 END,
+            CASE WHEN vhnd_conducted is not null and vhnd_conducted>0 THEN 1 ELSE 0 END,
+            CASE WHEN cbe_conducted is not null and cbe_conducted>1 THEN 1 ELSE 0 END,
             thr_v2.thr_distribution_image_count,
             0,
             0,
             0,
             0,
             0
             FROM awc_location_local awc_location
-            LEFT JOIN "{cbe_table}" cbe_table on  awc_location.doc_id = cbe_table.awc_id
-            LEFT JOIN "{vhnd_table}" vhnd_table on awc_location.doc_id = vhnd_table.awc_id
+            LEFT JOIN (
+                        select awc_id,
+                               count(*) as cbe_conducted from
+                            "{cbe_table}"
+                                WHERE date_trunc('MONTH', date_cbe_organise) = %(start_date)s
```

Comment:
In general, functions shouldn't be included in where clauses. They need special indexes to allow the query planner to efficiently filter them. These queries would be better written as `date_cbe_organise >= %(start_date)s AND date_cbe_organize < %(next_month_date)s`. 

With that said, it looks like these were doing sequential scans of the table before this change, so its not likely to make an immediate impact. How large are these tables expected to be? I think these sequential scans will be an issue at some point.

(this comment also applies to the VHND table below)

Reply:
Oww. I did not know about the function thing. yeah I will change it.
About the size of table cbe and vhnd are pretty small we get 0, 1 or 2 forms per awc per moth. Do you think thats going to be too big? If so, if I create an intermediate agg table for these two and then make join when those table will that solve it?

Reply:
At full scale, that's going to be 2 million forms every month. I think we'll just want an index on this column at some point. I don't know that its worth it to create an intermediate table (though it may be good to do that to keep with the current workflow)

Reply:
@Rohit25negi are you still planning to change that to not use the function?